src/vs/editor/contrib/referenceSearch/common/referenceSearch.ts

```diff
@@ -39,4 +39,30 @@ export function provideReferences(model: IReadOnlyModel, position: Position, pro
 	});
 }
 
+export function provideExternalReferences(model: IReadOnlyModel, query: ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: IReferenceInformation[]) => void): TPromise<IReferenceInformation[]> {
+
+	// collect references from all providers
+	const promises = ExternalReferenceProviderRegistry.ordered(model).map(provider => {
+		return asWinJsPromise((token) => {
+			return provider.provideExternalReferences(query, hints, token, progress);
+		}).then(result => {
+			if (Array.isArray(result)) {
```

Comment:
I would always adapt the typing to reflect this so tsc actually forces you to do these checks. But I guess forcing a good coding style on the VS Code codebase is out of scope